src/argilla/server/apis/v1/handlers/workspaces.py

```diff
@@ -44,3 +44,15 @@ async def get_workspace(
         )
 
     return workspace
+
+
+@router.get("/me/workspaces", response_model=Workspaces)
+async def list_workspaces_me(
+    *,
+    db: AsyncSession = Depends(get_async_db),
+    current_user: User = Security(auth.get_current_user),
+) -> Workspaces:
+    await authorize(current_user, WorkspacePolicyV1.list_workspaces_me)
+
+    workspaces = await accounts.list_workspaces_by_user_id(db, current_user.id)
+    return Workspaces(items=workspaces)
```

Comment:
```suggestion
    await authorize(current_user, WorkspacePolicyV1.list_workspaces_me)

    if current_user.is_owner:
        workspaces = await accounts.list_workspaces(db)
    else:
        workspaces = await accounts.list_workspaces_by_user_id(db, current_user.id)
    return Workspaces(items=workspaces)
```

Reply:
Thanks for the suggestion (sorry I forgot to include you as co-author 😞)